server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -990,8 +990,21 @@ public final void parse(String name, ParserContext parserContext, Map<String, Ob
                         iterator.remove();
                         continue;
                     }
-                    throw new MapperParsingException("unknown parameter [" + propName
-                        + "] on mapper [" + name + "] of type [" + type + "]");
+                    if (parserContext.isFromDynamicTemplate() && parserContext.indexVersionCreated().before(Version.V_8_0_0)) {
```

Comment:
I don't think we can have a dynamic template in an index >8.0.0 with a malformed parameter since we cannot add it, but I put the version check here regardless for safety. @romseygeek wdyt?

Reply:
++